src/store/sagas/positions.ts

```diff
@@ -254,23 +97,27 @@ function* handleInitPositionWithAZERO(action: PayloadAction<InitPositionData>):
       true
     )
 
-    let azeroAmount = 0n
-    if (tokenX === TESTNET_WAZERO_ADDRESS) {
-      azeroAmount = azeroBalance > xAmountWithSlippage ? xAmountWithSlippage : azeroBalance
-    } else {
-      azeroAmount = azeroBalance > yAmountWithSlippage ? yAmountWithSlippage : azeroBalance
+    if (
+      (tokenX === wazeroAddress && tokenXAmount !== 0n) ||
+      (tokenY === wazeroAddress && tokenYAmount !== 0n)
+    ) {
+      let azeroAmount = 0n
+      if (tokenX === wazeroAddress) {
+        azeroAmount = azeroBalance > xAmountWithSlippage ? xAmountWithSlippage : azeroBalance
```

Comment:
You are actually doubling the logic of getting the token with slippage or balance separately for the x and y tokens. I suggest something like this:

```Typescript
  const isTokenX = tokenX === wazeroAddress;
  const slippageAmount = isTokenX ? xAmountWithSlippage : yAmountWithSlippage;
  const azeroAmount = azeroBalance > slippageAmount ? slippageAmount : azeroBalance;

  const depositTx = wazero.depositTx(azeroAmount, WAZERO_DEPOSIT_OPTIONS);
  txs.push(depositTx);
```